### PR TITLE
EPMDJ-2888: Create new Card component. Update Tests2code page layout

### PR DESCRIPTION
**Pull request by RomanAchilov**
Issue: EPMDJ-2888:

### CODE REVIEW

File: src/components/card/card.tsx [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+}
+
+const card = BEM(styles);
+
+export const Card = card(({
+  className, cardData: {
+    totalCount = 0, coveredCount = 0, type, stats = 0,
+  },
+}: Props) => {
+  const isMethods = ['All METHODS', 'NEW', 'MODIFIED', 'DELETED'].includes(type);
```
Comment (RomanDavlyatshin) [inline src/components/card/card.tsx:26]: Remove convoluted logic from `Card` component
Make base card component and add 3 props (stats suffix, unit, progress bar units? (think of a better name)

File: src/components/card/card.tsx [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+}
+
+interface Props {
+  className?: string;
+  cardData: CardData;
+}
+
+const card = BEM(styles);
+
+export const Card = card(({
```
Comment (RomanDavlyatshin) [inline src/components/card/card.tsx:21]: Think of a better name (`Card` is not generic at all, maybe StatsCard or smth?)

File: src/pages/agent-full-page/coverage-plugin/project-methods-card/project-methods-card.tsx [hunk 3]
```diff
@@ -20,63 +18,23 @@ export const ProjectMethodsCard = projectMethodsCard(
...
-                  || 0} deleted methods were covered in previous build.`}
-            </DeletedMethodsAdditionalInfo>
-          )}
-          type="deleted"
-        />
-      </Card>
-    </div>
-  ),
+  }: Props) => {
+    const cardMethods = [
```
Comment (pavyarov) [inline src/pages/agent-full-page/coverage-plugin/project-methods-card/project-methods-card.tsx:24]: dunno. i think this must be JSX

File: src/pages/agent-full-page/coverage-plugin/project-methods-card/project-methods-card.tsx [hunk 4]
```diff
@@ -20,63 +20,55 @@ export const ProjectMethodsCard = projectMethodsCard(
...
+          label="NEW"
+          covered={newMethods?.covered || 0}
+          totalCount={newMethods?.total || 0}
+          progress={convertToPercentage((newMethods?.covered || 0), (newMethods?.total || 0))}
+        >
+          {Boolean(risks?.new) && (
+            <LinkButton size="small" onClick={() => setRisksFilter('new')} data-test="info-panel:link-button:new:risks">
+              {risks?.new} risks
+            </LinkButton>
+          ) }
```
Comment (pavyarov) [inline src/pages/agent-full-page/coverage-plugin/project-methods-card/project-methods-card.tsx:45]:
```suggestion
          )}

```

File: src/components/info-panel/info-panel.tsx [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+
+import styles from './info-panel.module.scss';
+
+interface Props {
+  className?: string;
+  label: string;
+  children?: React.ReactNode;
+  covered: number | string;
+  totalCount: number;
+  progress?: number;
```
Comment (pavyarov) [inline src/components/info-panel/info-panel.tsx:13]: u have `covered` and `totalCount` any reason to pass `progress`

File: src/components/info-panel/info-panel.tsx [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+import { BEM } from '@redneckz/react-bem-helper';
+import { Panel } from '@drill4j/ui-kit';
+
+import styles from './info-panel.module.scss';
+
+interface Props {
+  className?: string;
+  label: string;
+  children?: React.ReactNode;
+  covered: number | string;
```
Comment (pavyarov) [inline src/components/info-panel/info-panel.tsx:11]: `number | string` ?

File: src/common/style-guide/_colors.scss [hunk 7]
```diff
@@ -78,3 +78,16 @@ $orange: (
   medium-tint: #f7b84f,
   light-tint: #fef6e9,
 );
+
+$data-visualization: (
+  dv1: #67d5b5,
```
Comment (pavyarov) [inline src/common/style-guide/_colors.scss:83]: dunno, still does't like this names

File: src/components/info-panel/info-panel.tsx (hunk 6 above)
Comment (RomanAchilov) [reply src/components/info-panel/info-panel.tsx:11]: tests tab
covered={`${count}%`}
image

File: src/components/info-panel/info-panel.tsx [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+  totalCount: number;
+  progress?: number;
+}
+
+const infoPanel = BEM(styles);
+
+export const InfoPanel = infoPanel(({
```
[the remark this answers is not in the document]
Comment (RomanAchilov) [reply src/components/info-panel/info-panel.tsx:18]: InfoCard?

File: src/components/info-card/info-card.tsx [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+  covered?: number | string;
+  totalCount?: number;
+}
+
+const infoCard = BEM(styles);
+
+export const InfoCard = infoCard(({
+  className, label, children, covered = 0, totalCount = 0,
+}: Props) => {
+  const coverage = typeof covered === 'string' ? covered : `${(covered / totalCount) * 100}%`;
```
Comment (pavyarov) [inline src/components/info-card/info-card.tsx:20]: so if covered = 0 ?)
`const coverage = 'NaN%'` 😆

**Pull request merged**